app/(product)/public-cloud/(product)/aws-roles/[licencePlate]/[role]/page.tsx

```diff
@@ -0,0 +1,68 @@
+'use client';
+
+import TableAWSRoles from '@/components/table/TableAWSRoles';
+import { User } from '@/app/api/public-cloud/aws-roles/helpers';
+import TableBodyAWSRoles from '@/components/table/TableBodyAWSRoles';
+import { capitalizeFirstLetter } from '@/components/utils/capitalizeFirstLetter';
+import { useQuery } from '@tanstack/react-query';
+import { usePathname } from 'next/navigation';
+
+const pathParamRoleToRole = (pathRole: string): string => {
+  const role = capitalizeFirstLetter(pathRole.replace(/-/g, ' ').slice(0, -1));
+  return role;
+};
+
+const parseUrlParams = (
+  url: string,
+): { licencePlate: string; userRole: string; currentPage: string; pageSize: string } => {
+  const licencePlate = url.split('/')[3];
+  const userRole = pathParamRoleToRole(url.split('/')[4]);
+  const currentPage = url.split('?')[1]?.split('&')[0].split('=')[1] || '1';
+  const pageSize = url.split('?')[1]?.split('&')[1].split('=')[1] || '5';
+
+  return { licencePlate, userRole, currentPage, pageSize };
+};
+
+async function GetUsersPaginatedList(
+  licencePlate: string,
+  role: string,
+  currentPage: string,
+  pageSize: string,
+): Promise<any> {
+  const res = await fetch(
+    `/api/public-cloud/aws-roles/getUsersList?licencePlate=${licencePlate}&role=${role}&page=${currentPage}&pageSize=${pageSize}`,
+  );
+  if (!res.ok) {
+    throw new Error('Network response was not ok for fetch user image');
+  }
+  const data = await res.json();
+  return data;
+}
+
+export default function ProductAWSRoles() {
+  const pathName = usePathname();
+  const { licencePlate, userRole, currentPage, pageSize } = parseUrlParams(pathName);
```

Comment:
What don't we use `useParams` to get `path params` and use `useSearchParams` to get `query params` in client side?
- https://nextjs.org/docs/app/api-reference/functions/use-params
- https://nextjs.org/docs/app/api-reference/functions/use-search-params

Also, take a look at the helper function `parsePaginationParams` in the our codebase:
- https://github.com/bcgov/platform-services-registry/blob/main/app/zap/results/page.tsx#L141

Reply:
changed, thank you!